hpx/parallel/spmd_block.hpp

```diff
@@ -84,34 +87,47 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v2)
            barrier_.get().wait();
         }
 
+        hpx::future<void> sync_all(hpx::launch::async_policy const &) const
+        {
+           return barrier_.get().wait(hpx::launch::async);
+        }
+
     private:
         std::size_t num_images_;
         std::size_t image_id_;
-        mutable std::reference_wrapper<hpx::lcos::local::barrier> barrier_;
+        mutable std::reference_wrapper<hpx::lcos::barrier> barrier_;
     };
 
     namespace detail
     {
         template <typename F>
         struct spmd_block_helper
         {
-            mutable std::reference_wrapper<hpx::lcos::local::barrier> barrier_;
-            typename std::decay<F>::type f_;
+            std::string name_;
             std::size_t num_images_;
 
             template <typename ... Ts>
             void operator()(std::size_t image_id, Ts && ... ts) const
             {
-                spmd_block block(num_images_, image_id, barrier_);
+                hpx::lcos::barrier
+                    barrier(name_ + "_barrier" , num_images_, image_id);
```

Comment:
With commit 23ce7fe33d8afdc528631c85236d18d0e31bc94c. I'm now having an exception that give me more informations: terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<hpx::exception> >'
  what():  bind: Address already in use: HPX(network_error)``